pyisemail/validators/dns_validator.py

```diff
@@ -37,8 +37,15 @@ def is_valid(self, domain, diagnose=False):
         # we will raise a warning because we didn't immediately find an MX
         # record.
         try:
-            dns.resolver.resolve(domain, MX)
+            records = dns.resolver.resolve(domain, MX)
             dns_checked = True
+
+            # Even if there's an MX record set we need to verify the preference
+            # value. If it's value is 0 it should return null MX.
+            if len(records) > 0:
+                for rdata in records:
+                    if rdata.preference == 0:
```

Comment:
**issue:** Don't we need to check for a zero length (i.e. `.`) record in addition to the zero preference? My reading of [Section 3 of the RFC](https://www.rfc-editor.org/rfc/rfc7505.html#section-3) says yes. Particularly combined with [the SMTP fallback behavior](https://datatracker.ietf.org/doc/html/rfc5321#section-5.1) of, "If an empty list of MXs is returned, the address is treated as if it was associated with an implicit MX RR, with a preference of 0, pointing to that host," and [the examples at the end of RFC 974](https://datatracker.ietf.org/doc/html/rfc974), it seems like zero does not imply by itself that it's a null MX record.

A test of the behavior would be good to show that it works as intended.